plenum/server/replica.py

```diff
@@ -913,19 +914,20 @@ def _process_valid_preprepare(self, pre_prepare, sender):
             self.discard(pre_prepare, 'node is not participating',
                          self.logger.info)
             return None
-        old_state_root = self.stateRootHash(pre_prepare.ledgerId, to_str=False)
+        pre_state_root = self.stateRootHash(pre_prepare.ledgerId, to_str=False)
         why_not_applied = self._apply_pre_prepare(pre_prepare, sender)
         if why_not_applied is not None:
             return why_not_applied
         self.addToPrePrepares(pre_prepare)
         if self.isMaster:
-            # TODO: can old_state_root be used here instead?
+            # TODO: can pre_state_root be used here instead?
             state_root = self.stateRootHash(pre_prepare.ledgerId, to_str=False)
             self.node.onBatchCreated(pre_prepare.ledgerId, state_root)
-            # BLS multi-sig:
-            self._bls_bft_replica.process_pre_prepare(pre_prepare, sender)
-            self.logger.trace("{} saved shared multi signature for root".format(self, old_state_root))
-        self.trackBatches(pre_prepare, old_state_root)
+        # BLS multi-sig:
+        self._bls_bft_replica.process_pre_prepare(pre_prepare, sender)
```

Comment:
Saving of BLS multi-sig needs to be done on master for master only, so please revert it.
Verification of BLS multi-sig is done for all instances.

Reply:
Done